Ical.Net/Utility/CollectionHelpers.cs

```diff
@@ -66,18 +62,21 @@ public static bool Equals<T>(IEnumerable<T> left, IEnumerable<T> right, bool ord
             return left.SequenceEqual(right);
         }
 
+        // No multiple enumerations
+        var rightArray = right as T[] ?? right.ToArray();
+        var leftArray = left as T[] ?? left.ToArray();
         try
         {
             //Many things have natural IComparers defined, but some don't, because no natural comparer exists
-            return left.OrderBy(l => l).SequenceEqual(right.OrderBy(r => r));
+            return leftArray.OrderBy(l => l).SequenceEqual(rightArray.OrderBy(r => r));
         }
         catch (Exception)
         {
             //It's not possible to sort some collections of things (like Calendars) in any meaningful way. Properties can be null, and there's no natural
             //ordering for the contents therein. In cases like that, the best we can do is treat them like sets, and compare them. We don't maintain
             //fidelity with respect to duplicates, but it seems better than doing nothing
-            var leftSet = new HashSet<T>(left);
-            var rightSet = new HashSet<T>(right);
+            var leftSet = new HashSet<T>(leftArray);
```

Comment:
I think this whole way of comparing isn't too helpful. We discussed this in another ticket, that `Equals` might not be too helpful for many of the types, e.g. Calendar. But that's a topic for a different PR.

Reply:
Yes, I wanted to stick to NRT